website/pages/blog/setting-up-graphql-server-to-scale-with-graphql-codegen-server-preset.mdx

```diff
@@ -0,0 +1,166 @@
+
+On top of that, all datasources and resolvers must be put together into the GraphQL server. In this case, it's the `server.ts` file. Who owns this file and other maintenance of the server such as `codegen.yml` (we all use GraphQL Codegen right? 😉), package updates, security patches, etc.?
+
+### Solution: Split into Modules
+
+It is fairly common to see teams change name or mission. However, the thing that does not change very often (and in some cases, never!) is business domain. If we split our code based on business domain and assign teams accordingly, it becomes much more scalable.
+
+On the maintenance of the server, it is best to have a small group of dedicated maintainers. This could consist of one member from each team in the codebase and membership can rotate, or another team altogether. Having a dedicated group of maintainers help reduce noise and cognitive load to the rest of the teams.
+
+Using the same example before, we can identify 3 main domains: User, Auth and Book, each have CODEOWNERS set up to notify them of changes. The Maintainers own `server.ts` and other configs.
+
+```markdown
+├── src/
+│   ├── schema/
+│   │   ├── user/ # CODEOWNERS set up to ping TeamA if anything changes in this folder
+│   │   │   ├── datasources/
+│   │   │   │   ├── UserDatasource.ts
+│   │   │   ├── resolvers/
+│   │   │   │   ├── userResolvers.ts
+│   │   │   ├── userSchema.graphql
+│   │   ├── auth/ # CODEOWNERS set up to ping TeamA if anything changes in this folder
+│   │   │   ├── datasources/
+│   │   │   │   ├── AuthDatasource.ts
+│   │   │   ├── resolvers/
+│   │   │   │   ├── authResolvers.ts
+│   │   │   ├── authSchema.graphql
+│   │   ├── book/ # CODEOWNERS set up to ping TeamB if anything changes in this folder
+│   │   │   ├── datasources/
+│   │   │   │   ├── BookDatasource.ts
+│   │   │   ├── resolvers/
+│   │   │   │   ├── bookResolvers.ts
+│   │   │   ├── bookSchema.graphql
+│   ├── server.ts   # CODEOWNERS set up to ping Maintainers if this changes
+│   ├── codegen.yml # CODEOWNERS set up to ping Maintainers if this changes
+```
+
+## The Hard Problems
+
+### 1. How to Enforce Best Practices for All Teams
+
+*   Thanks to GraphQL Codegen, we can generate types for resolvers
+*   However, there's no stopping teams from *NOT* using the types
+
+### 2. How to Minimise Noise to the Core Maintainers
+
+*   Core maintainer of the GraphQL server usually need to manage many things:
+    *   `codegen.yml`
+    *   Main server logic e.g. putting the resolvers together (show an example with GraphQL Yoga)
+
+*   Resolver Problm: adding a new schema module requires wiring up to the main server file/s which requires core maintainer's attention at least once:
```

Comment:
```suggestion
*   Resolver problem: adding a new schema module requires wiring up to the main server file/s which requires core maintainer's attention at least once:
```

Reply:
Thanks for catching that! This bit is now expanded and hopefully has no spelling mistakes 😆